guide/src/class/numeric.md

```diff
@@ -0,0 +1,373 @@
+# Emulating numeric types
```

Comment:
I really like how this new page reads. It tells a nice story along the lines of "implementing your first PyO3 class". 

That said, the existing `class.md` is somewhere between a reference and a story itself. I'd love it if we could make the distinction clearer in splitting out the reference bits of `class.md` into a `#[pyclass]` attribute reference and then have a whole "getting started with PyO3 classes" story (of which this page is one part).

Reply:
I remember being a bit annoyed with this myself, when trying to figure out how to make something work I had to go back and forth between the book and the api documentation which was quite annoying.

> splitting out

I think a bit of duplication is also fine, it doesn't need to be a hard split.

Reply:
Agreed its impossible to avoid duplication; I guess what I meant by splitting out is that the reference portion would always contain full detail, the story might introduce the key elements and then just link to the reference portion for readers who want to see the whole lot.